controller/organization.go

```diff
@@ -0,0 +1,157 @@
+
+		// Lookup the identity for the current user
+		userIdentity, err := appl.Identities().Load(ctx, *currentUser)
+		if err != nil {
+			return errors.NewInternalErrorFromString(ctx, "Error looking up current user")
+		}
+
+		// Lookup the organization resource type
+		resourceType, err := appl.ResourceTypeRepository().Lookup(ctx, account.IDENTITY_RESOURCE_TYPE_ORGANIZATION)
+		if err != nil {
+			return errors.NewInternalErrorFromString(ctx, "Error looking up resource type 'identity/organization'")
+		}
+
+		// Create the organization resource
+		res := &resource.Resource{
+			Name:           *ctx.Payload.Name,
+			ResourceType:   *resourceType,
+			ResourceTypeID: resourceType.ResourceTypeID,
+		}
+
+		err = appl.ResourceRepository().Create(ctx, res)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
+		}
+
+		// Create the organization identity
+		orgIdentity := &account.Identity{
+			IdentityResourceID: &res.ResourceID,
+		}
+
+		err = appl.Identities().Create(ctx, orgIdentity)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
+		}
+
+		organizationId = orgIdentity.ID
+
+		// Lookup the identity/organization owner role
+		ownerRole, err := appl.RoleRepository().Lookup(ctx, ORGANIZATION_OWNER_ROLE, account.IDENTITY_RESOURCE_TYPE_ORGANIZATION)
+
+		if err != nil {
+			return errors.NewInternalErrorFromString(ctx, "Error looking up owner role for 'identity/organization' resource type")
+		}
+
+		// Assign the owner role for the new organization to the current user
+		identityRole := &role.IdentityRole{
+			IdentityID: userIdentity.ID,
+			ResourceID: res.ResourceID,
+			RoleID:     ownerRole.RoleID,
+		}
+
+		err = appl.IdentityRoleRepository().Create(ctx, identityRole)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
+		}
+
+		return err
+	})
+
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	log.Debug(ctx, map[string]interface{}{
+		"organization_id": organizationId.String(),
+	}, "organization created")
+
+	orgId := organizationId.String()
+
+	return ctx.Created(&app.CreateOrganizationResponse{&orgId})
+}
+
+// List runs the list action.
+func (c *OrganizationController) List(ctx *app.ListOrganizationContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+
+	var orgs []account.IdentityOrganization
+
+	err = application.Transactional(c.db, func(appl application.Application) error {
+
+		orgs, err = appl.Identities().ListOrganizations(ctx, *currentUser)
+
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
+		}
+
+		return err
+	})
+
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	results := []*app.OrganizationData{}
+
+	for _, org := range orgs {
+		orgData := &app.OrganizationData{
```

Comment:
Could you move this logic out to a `func toAppOrganization(..)` like the way we do in other places https://github.com/fabric8-services/fabric8-auth/blob/master/controller/collaborators.go#L108

Reply:
I've moved this out to a separate function as suggested